### PR TITLE
bug(spec-specs, tests): EIP-8037 strict block-gas inclusion rule

### PR DESCRIPTION
**Pull request by chfast**
## 🗒️ Description

EIP-8037 specifies the per-dimension inclusion check at block start as:

    min(TX_MAX_GAS_LIMIT, tx.gas) <= regular_gas_available
    tx.gas <= state_gas_available

EELS Amsterdam was subtracting intrinsic.state / intrinsic.regular from tx.gas in each dimension, which let a transaction with non-zero state intrinsic (CREATE, EIP-7702 set_code) be included when tx.gas exceeded block.gas_limit by up to intrinsic.regular (state-dim) or intrinsic.state (regular-dim). evmone, revm, and erigon enforce the EIP literally and reject; pre-bump geth rejects too.

Add a parametrized test under EIP-8037 covering the boundary across all post-EIP-7702 tx types and contract-creation kinds, at two block-gas-limit values to avoid hard-coded-cap false-positives.

## 🔗 Related Issues or PRs

N/A.

## ✅ Checklist

- [x] All: Ran fast static checks to avoid unnecessary CI fails, see also [Code Standards](https://eest.ethereum.org/main/getting_started/code_standards/) and [Enabling Pre-commit Checks](https://eest.ethereum.org/main/dev/precommit/):
    ```console
    just static
    ```
- [x] All: PR title adheres to the [repo standard](https://eest.ethereum.org/main/getting_started/contributing/?h=contri#commit-messages-issue-and-pr-titles) - it will be used as the squash commit message and should start `type(scope):`.
- [x] All: Considered updating the online docs in the [./docs/](/ethereum/execution-specs/blob/HEAD/docs/) directory.
- [x] All: Set appropriate labels for the changes (only maintainers can apply labels).
- [x] Tests: Ran `mkdocs serve` locally and verified the auto-generated docs for new tests in the [Test Case Reference](https://eest.ethereum.org/main/tests/) are correctly formatted.
- [x] Tests: For PRs implementing a missed test case, update the [post-mortem document](/ethereum/execution-specs/blob/HEAD/docs/writing_tests/post_mortems.md) to add an entry the list.

### CODE REVIEW

**Comment by marioevz** [comment]
We are in the process of merging EIP-8037 to `forks/amsterdam` via this PR: https://github.com/ethereum/execution-specs/pull/2901

Once the PR is merged we can point this to the `forks/amsterdam` branch and then merge.

**Review by misilva73 — COMMENTED** [review]
I think the correct check is the current EIP version and this PR aligns the execution specs with the EIP. So I think we should merge it.

We discussed this some time ago and decided to do this simpler check instead of the one currently in EELS

File: src/ethereum/forks/amsterdam/fork.py [hunk 1]
```diff
@@ -569,17 +564,12 @@ def check_transaction(
...
     blob_gas_available = MAX_BLOB_GAS_PER_BLOCK - block_output.blob_gas_used
 
-    # Worst-case regular contribution: tx.gas minus the portion that
-    # must go to intrinsic state gas, capped at TX_MAX_GAS_LIMIT.
-    worst_case_regular = min(TX_MAX_GAS_LIMIT, tx.gas - intrinsic.state)
-    if worst_case_regular > regular_gas_available:
-        raise GasUsedExceedsLimitError("regular gas used exceeds limit")
+    # EIP-8037 per-dimension inclusion check.
+    if min(TX_MAX_GAS_LIMIT, tx.gas) > regular_gas_available:
+        raise GasUsedExceedsLimitError("gas used exceeds limit")
```
Comment (spencer-tb) [inline src/ethereum/forks/amsterdam/fork.py]:
```suggestion
        raise GasUsedExceedsLimitError("regular gas used exceeds limit")
```

File: src/ethereum/forks/amsterdam/fork.py [hunk 2]
```diff
@@ -569,17 +564,12 @@ def check_transaction(
...
 
-    # Worst-case state contribution: tx.gas minus the portion that
-    # must go to intrinsic regular gas.
-    worst_case_state = tx.gas - intrinsic.regular
-    if worst_case_state > state_gas_available:
-        raise GasUsedExceedsLimitError("state gas used exceeds limit")
+    if tx.gas > state_gas_available:
+        raise GasUsedExceedsLimitError("gas used exceeds limit")
```
Comment (spencer-tb) [inline src/ethereum/forks/amsterdam/fork.py]:
```suggestion
        raise GasUsedExceedsLimitError("state gas used exceeds limit")
```

**Review by spencer-tb — APPROVED** [review]
Thanks! LGTM. Adding 2 small nits to preserve state/regular in exception message.

**Comment by spencer-tb** [comment]
Will merge once CI passes! Thx :)
